tests/js/server/dump/dump-mixed.js

```diff
@@ -0,0 +1,553 @@
+
+      assertEqual("skiplist", c.getIndexes()[4].type);
+      assertTrue(c.getIndexes()[4].unique);
+      assertFalse(c.getIndexes()[4].sparse);
+      assertEqual([ "a_su" ], c.getIndexes()[4].fields);
+
+      assertEqual("hash", c.getIndexes()[5].type);
+      assertFalse(c.getIndexes()[5].unique);
+      assertTrue(c.getIndexes()[5].sparse);
+      assertEqual([ "a_hs1", "a_hs2" ], c.getIndexes()[5].fields);
+
+      assertEqual("skiplist", c.getIndexes()[6].type);
+      assertFalse(c.getIndexes()[6].unique);
+      assertTrue(c.getIndexes()[6].sparse);
+      assertEqual([ "a_ss1", "a_ss2" ], c.getIndexes()[6].fields);
+
+      assertFalse(c.getIndexes()[7].unique);
+      assertEqual("fulltext", c.getIndexes()[7].type);
+      assertEqual([ "a_f" ], c.getIndexes()[7].fields);
+
+      assertEqual("geo", c.getIndexes()[8].type);
+      assertEqual([ "a_la", "a_lo" ], c.getIndexes()[8].fields);
+      assertFalse(c.getIndexes()[8].unique);
+
+      assertEqual(0, c.count());
+    },
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test truncate
+////////////////////////////////////////////////////////////////////////////////
+
+    testTruncated : function () {
+      var c = db._collection("UnitTestsDumpTruncated");
+      var p = c.properties();
+
+      assertEqual(2, c.type()); // document
+      assertFalse(p.waitForSync);
+
+      assertEqual(1, c.getIndexes().length); // just primary index
+      assertEqual("primary", c.getIndexes()[0].type);
+      assertEqual(0, c.count());
+    },
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test autoinc keygen
+////////////////////////////////////////////////////////////////////////////////
+//
+//    testKeygenAutoInc : function () {
+//      var c = db._collection("UnitTestsDumpKeygen");
+//      var p = c.properties();
+//
+//      assertEqual(2, c.type()); // document
+//      assertFalse(p.waitForSync);
+//      assertEqual("autoincrement", p.keyOptions.type);
+//      assertEqual(42049, p.keyOptions.lastValue);
+//      assertFalse(p.keyOptions.allowUserKeys);
+//      assertEqual(7, p.keyOptions.offset);
+//      assertEqual(42, p.keyOptions.increment);
+//
+//      assertEqual(1, c.getIndexes().length); // just primary index
+//      assertEqual("primary", c.getIndexes()[0].type);
+//      assertEqual(1001, c.count());
+//
+//      for (var i = 0; i < 1000; ++i) {
+//        var doc = c.document(String(7 + (i * 42)));
+//
+//        assertEqual(String(7 + (i * 42)), doc._key);
+//        assertEqual(i, doc.value);
+//        assertEqual({ value: [ i, i ] }, doc.more);
+//      }
+//      doc = c.save({});
+//      assertEqual(doc._key, "42091");
+//    },
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test padded keygen
+////////////////////////////////////////////////////////////////////////////////
+
+    testKeygenPadded : function () {
```

Comment:
The same (or almost the same) method I've just reviewed in "tests/js/server/dump/dump-mixed-cluster.js". 
Please re-use that method here. This is hard to maintain.